src/library/scala/collection/StringOps.scala

```diff
@@ -1201,14 +1201,16 @@ final class StringOps(private val s: String) extends AnyVal {
   def withFilter(p: Char => Boolean): StringOps.WithFilter = new StringOps.WithFilter(p, s)
 
   /** The rest of the string without its first char.
-    * @note $unicodeunaware
-    */
-  def tail: String = slice(1, s.length)
+   *  @throws NoSuchElementException if the string is empty.
+   *  @note $unicodeunaware
+   */
+  def tail: String = if(s.isEmpty) throw new NoSuchElementException("tail of empty String") else slice(1, s.length)
```

Comment:
Most collections follow `IterableOps` in throwing `UnsupportedOperation` (but `NSE` for `head` and `last`).

A couple deviate and throw `NSE`. It's fine to leave it as it is here.

Example showing difficulty of enforcing uniformity of behavior:
```
scala> 0.until(0).tail
java.util.NoSuchElementException: tail on empty Range
  at scala.collection.immutable.Range$.scala$collection$immutable$Range$$emptyRangeError(Range.scala:631)
  at scala.collection.immutable.Range.tail(Range.scala:144)
  ... 30 elided

scala> 0.to(0).tail
val res4: scala.collection.immutable.Range = empty Range 0 until 0
```
example with descriptive message:
```
scala> collection.immutable.ArraySeq(Array(42)).tail.tail
java.lang.UnsupportedOperationException: tail of empty array
  at scala.collection.ArrayOps$.tail$extension(ArrayOps.scala:346)
  at scala.collection.immutable.ArraySeq.tail(ArraySeq.scala:242)
  ... 30 elided
```

Reply:
I meant to add, I'll merge as-is if you don't change it. Merges are paused until 2.13.15 is released, I assume.